From c5f29721319f169389daf55ae1c47b4d592dd6b8 Mon Sep 17 00:00:00 2001
From: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
Date: Mon, 9 Nov 2020 15:45:05 +0530
Subject: [PATCH 1748/1851] dt-bindings: regulator: DA9121: Add DA9121 bindings

commit ae2af212053ae760582d15c0b8d14e520b4b17cd from
https://github.com/Xilinx/linux-xlnx.git

Document the device tree bindings of dialog semiconductor's
DA9121 family of devices.

Signed-off-by: Raviteja Narayanam <raviteja.narayanam@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../devicetree/bindings/regulator/da9121.txt  | 120 ++++++++++++++++++
 .../regulator/dlg,da9121-regulator.h          |  22 ++++
 2 files changed, 142 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/regulator/da9121.txt
 create mode 100644 include/dt-bindings/regulator/dlg,da9121-regulator.h

diff --git a/Documentation/devicetree/bindings/regulator/da9121.txt b/Documentation/devicetree/bindings/regulator/da9121.txt
new file mode 100644
index 000000000000..0af3ae45817a
--- /dev/null
+++ b/Documentation/devicetree/bindings/regulator/da9121.txt
@@ -0,0 +1,120 @@
+* Dialog Semiconductor DA9121 Single-channel 10A double-phase buck converter
+* Dialog Semiconductor DA9122 Double-channel  5A single-phase buck converter
+* Dialog Semiconductor DA9220 Double-channel  3A single-phase buck converter
+* Dialog Semiconductor DA9217 Single-channel  6A double-phase buck converter
+* Dialog Semiconductor DA9130 Single-channel 10A double-phase buck converter
+* Dialog Semiconductor DA9131 Double-channel  5A single-phase buck converter
+* Dialog Semiconductor DA9132 Double-channel  3A single-phase buck converter
+
+Required properties:
+
+- compatible : Should be "dlg,da9121" for DA9121
+- compatible : Should be "dlg,da9122" for DA9122
+- compatible : Should be "dlg,da9220" for DA9220
+- compatible : Should be "dlg,da9217" for DA9217
+- compatible : Should be "dlg,da9130" for DA9130
+- compatible : Should be "dlg,da9131" for DA9131
+- compatible : Should be "dlg,da9132" for DA9132
+- reg : Specifies the I2C slave address.
+- interrupt-parent : Specifies the reference to the interrupt controller.
+- interrupts : IRQ line information.
+
+Current limits
+
+  This is PER PHASE, and the current limit setting in the devices reflect
+  that with a maximum 10A limit. Allowing for transients as high as 100%
+  of the rated current, this translates across the device range to per
+  channel figures as so:
+
+                               | DA9121   DA9122     DA9220     DA9217   DA9140
+                               | /DA9130  /DA9131    /DA9132
+    -----------------------------------------------------------------------------
+    Output current / channel   | 10000000   5000000   3000000   6000000  40000000
+    Output current / phase     |  5000000   5000000   3000000   3000000  10000000
+    Min regulator-min-microamp |  7000000   3500000   3500000   7000000  15000000
+    Max regulator-max-microamp | 20000000  10000000   6000000  12000000  80000000
+    Device hardware default    | 15000000   7500000   5500000  11000000  TBD
+
+Optional properties:
+
+- dlg,irq-polling-delay-passive : Specify the polling period, measured in
+  milliseconds, between interrupt status update checks.
+
+Sub-nodes:
+
+- regulators : This node defines the settings for the BUCK. The content of the
+  sub-node is defined by the standard binding for regulators; see regulator.txt.
+  The DA9121 regulator is bound using their names listed below:
+
+    buck1 : BUCK1
+    buck2 : BUCK2	//DA9122, DA9220, DA9131, DA9132 only
+
+Optional properties for regulators:
+
+- regulator-mode         : defined in include/dt-bindings/regulator/dlg,da9121-regulator.h
+- regulator-initial-mode : Same as regulator-mode
+- dlg,enable-gpio        : Specify a valid GPIO for platform control of the regulator.
+- dlg,ripple-cancel      : defined in include/dt-bindings/regulator/dlg,da9121-regulator.h
+                           Only present on multi-channel devices
+
+Example:
+	pmic: da9121@68 {
+		compatible = "dlg,da9121";
+		reg = <0x68>;
+
+		interrupt-parent = <&gpio6>;
+		interrupts = <11 IRQ_TYPE_LEVEL_LOW>;
+
+		dlg,irq-polling-delay-passive = <2000>;
+
+		regulators {
+			DA9121_BUCK1: buck1 {
+				regulator-name = "BUCK1";
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1900000>;
+				regulator-min-microamp = <7000000>;
+				regulator-max-microamp = <20000000>;
+				regulator-boot-on;
+				regulator-initial-mode = <DA9121_BUCK_MODE_AUTO>;
+				dlg,enable-gpio = <&gpio 1 GPIO_ACTIVE_HIGH>;
+			};
+		};
+	};
+
+
+Example:
+	pmic: da9122@68 {
+		compatible = "dlg,da9122";
+		reg = <0x68>;
+
+		interrupt-parent = <&gpio6>;
+		interrupts = <11 IRQ_TYPE_LEVEL_LOW>;
+
+		dlg,irq-polling-delay-passive = <2000>;
+
+		regulators {
+			DA9122_BUCK1: buck1 {
+				regulator-name = "BUCK1";
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1900000>;
+				regulator-min-microamp = <3500000>;
+				regulator-max-microamp = <10000000>;
+				regulator-boot-on;
+				regulator-initial-mode = <DA9121_BUCK_MODE_AUTO>;
+				dlg,enable-gpio = <&gpio6 1 GPIO_ACTIVE_HIGH>;
+				dlg,ripple-cancel = <DA9121_BUCK_RIPPLE_CANCEL_NONE>;
+			};
+			DA9122_BUCK2: buck2 {
+				regulator-name = "BUCK2";
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1900000>;
+				regulator-min-microamp = <3500000>;
+				regulator-max-microamp = <10000000>;
+				regulator-boot-on;
+				regulator-initial-mode = <DA9121_BUCK_MODE_AUTO>;
+				dlg,enable-gpio = <&gpio6 2 GPIO_ACTIVE_HIGH>;
+				dlg,ripple-cancel = <DA9121_BUCK_RIPPLE_CANCEL_NONE>;
+			};
+		};
+	};
+
diff --git a/include/dt-bindings/regulator/dlg,da9121-regulator.h b/include/dt-bindings/regulator/dlg,da9121-regulator.h
new file mode 100644
index 000000000000..c68d03e9b34f
--- /dev/null
+++ b/include/dt-bindings/regulator/dlg,da9121-regulator.h
@@ -0,0 +1,22 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+
+#ifndef _DT_BINDINGS_REGULATOR_DLG_DA9121_H
+#define _DT_BINDINGS_REGULATOR_DLG_DA9121_H
+
+/*
+ * These buck mode constants may be used to specify values in device tree
+ * properties (e.g. regulator-initial-mode).
+ * A description of the following modes is in the manufacturers datasheet.
+ */
+
+#define DA9121_BUCK_MODE_FORCE_PFM		0
+#define DA9121_BUCK_MODE_FORCE_PWM		1
+#define DA9121_BUCK_MODE_FORCE_PWM_SHEDDING	2
+#define DA9121_BUCK_MODE_AUTO			3
+
+#define DA9121_BUCK_RIPPLE_CANCEL_NONE		0
+#define DA9121_BUCK_RIPPLE_CANCEL_SMALL		1
+#define DA9121_BUCK_RIPPLE_CANCEL_MID		2
+#define DA9121_BUCK_RIPPLE_CANCEL_LARGE		3
+
+#endif
-- 
2.31.1

